pkg/registry/apis/provisioning/jobs/driver.go

```diff
@@ -228,16 +220,7 @@ func (d *jobDriver) processJob(ctx context.Context, job *provisioning.Job) error
 			return apifmt.Errorf("failed to get repository '%s': %w", job.Spec.Repository, err)
 		}
 
-		recorder := newJobProgressRecorder(d.onProgress(job))
-
-		err = worker.Process(ctx, repo, *job, recorder)
-		if err != nil {
```

Comment:
here was the problem. The Complete function can handle and should handle the error.